drivers/regmap/internal.h

```diff
@@ -0,0 +1,85 @@
+/****************************************************************************
+ * drivers/regmap/internal.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __DRIVERS_REGMAP_INTERNAL_H
+#define __DRIVERS_REGMAP_INTERNAL_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/regmap/regmap.h>
+#include <nuttx/mutex.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+typedef CODE void (*regmap_lock_t)(FAR void *);
+typedef CODE void (*regmap_unlock_t)(FAR void *);
+
+/* Configuration for the register map of a device.
+ * This structure is only used inside regmap.
+ */
+
+struct regmap_s
+{
+  mutex_t mutex;
```

Comment:
```suggestion
  mutex_t *mutex;
```
alloc the resource only if mutex lock is needed